### PR TITLE
Add support for elastic and layered seabed

### PR DESCRIPTION
**Pull request by mchitre**
**Key features**:
- Adds new elastic boundary condition (`ElasticBoundary`).
- Adds new multilayer boundary condition (`MultilayerElasticBoundary`).
- Adds keyword constructors for all boundary conditions for readability.
- Adds support for interfacial roughness on all boundary conditions (parameter `σ`).

**Bug fix**:
- Forces boundary condition parameters to floats. With integer boundary conditions, `PekerisModeSolver` had accuracy problems with group velocity estimation.

**Other changes**:
- Updates version number to `0.5.0`.
- Updates compat bounds on dep `Interpolations`.
- Improves docstrings with examples for boundary conditions.
- Updates relevant documentation (need to regenerate `html` pages later).
- Improves test coverage.

**Notes**:
- Interfacial roughness is not yet supported in Pekeris models.
- Shear speed is currently ignored for reflection coefficient computation.
- Only the top layer of multilayer bottoms is currently used for reflection coefficient computation.
- Roughness, shear speed and multilayer bottoms are supported in various `AcousticsToolbox` propagation models.

### CODE REVIEW

File: test/test_api_stdlib.jl [hunk 1]
```diff
@@ -202,16 +255,30 @@ end
...
   fld = @inferred SampledField([0.0 1.0; 1.0 2.0]; x=[0.0, 1.0], y=[0.0, 1.0])
+  @test fld isa UnderwaterAcoustics.SampledFieldXY
+  @test startswith(sprint(show, fld), "SampledField")
   @test @inferred is_range_dependent(fld)
   @test @inferred(fld(0.0, 0.0)) == 0.0
   @test @inferred(fld(0.5, 0.5)) == 1.0
   @test @inferred(fld(1.0, 1.0)) == 2.0
   @test @inferred(fld((x=0.0, y=0.0, z=0.0))) == 0.0
   @test @inferred(fld((x=0.5, y=0.5, z=0.0))) == 1.0
   @test @inferred(fld((x=1.0, y=1.0, z=0.0))) == 2.0
```
[the remark this answers is not in the document]
Comment (mchitre) [reply test/test_api_stdlib.jl:266]: Invalid comment

File: src/basic.jl [hunk 2]
```diff
@@ -122,6 +122,27 @@ function reflection_coef(θ, ρᵣ, cᵣ, δ=0.0)
...
+- relative shear sound speed of the reflecting medium to incidence medium `cᵣₛ`
+- dimensionless compressional absorption coefficient `δ`
+- dimensionless shear absorption coefficient `δₛ`
+
+WARNING: Fluid-solid reflection not implemented. Currently ignores shear.
+"""
+function reflection_coef(θ, ρᵣ, cᵣ, cᵣₛ, δ, δₛ)
+  if cᵣₛ != 0
+    # TODO: implement (current implementation ignores shear)
+    @warn "Fluid-solid reflection not implemented, ignoring shear..." maxlog=1
```
[the remark this answers is not in the document]
Comment (mchitre) [reply src/basic.jl:141]: Comment and warning already in place.

File: src/stdlib.jl [hunk 3]
```diff
@@ -8,40 +8,74 @@ export SandyGravel, CoarseSand, MediumSand, FineSand, VeryFineSand, ClayeySand
...
   δ::T
-  function FluidBoundary(ρ, c, δ)
+  σ::T
+  function FluidBoundary(ρ, c, δ, σ)
     ρ = in_units(u"kg/m^3", ρ)
     c = in_units(u"m/s", c)
-    isinf(c) && return new{typeof(c)}(zero(c), c, zero(c))
-    new{typeof(ρ)}(promote(ρ, c, δ)...)
+    σ = in_units(u"m", σ)
+    isinf(c) && return new{typeof(c)}(zero(c), c, zero(c), zero(c))
```
[the remark this answers is not in the document]
Comment (mchitre) [reply src/stdlib.jl:59]: We like boundary condition parameters to be floating point since they generally are.
